secure_message/repository/modifier.py

```diff
@@ -81,8 +81,29 @@ def add_unread(message, user):
     @staticmethod
     def mark_message_as_read(message, user):
         """Remove unread label from status"""
+        logger.debug("marking message as read with id", message=message['msg_id'])
+        # mark the message as read i.e. remove the unread label and set the `read at` time.
+        Modifier._mark_read(message, user)
+        # also need to mark any other messages in the thread as read
+        if message['thread_id']:
+            try:
+                conversation = Retriever.retrieve_thread(message['thread_id'], user)
```

Comment:
Do we have any way of only retrieving the messages that need acting on rather than pulling the whole thread? Understandable if we don't, but would be good to pull as little data as is necessary

Reply:
not presently afaik but we could add it?

Reply:
Up to you, its not a prerequisite for merging this but it might be a better way of doing things

Reply:
I think you're right, it should be a method on the Modifier. I'll take a look at how much work it would be to change. Ideally we'd tackle the whole service and make it more RESTful, it has a few oddities.

Reply:
I still have to pull the whole thread unfortunately, however I can change it to use a single db session to do all the work. I'll push up a change shortly.